apps/sms/views/conversation/js/conversation.js

```diff
@@ -692,11 +692,14 @@ var ConversationView = {
     // Add recipients and fill+focus the Compose area.
     var recipientPromise;
     if (activity.contact && number) {
-      recipientPromise = Utils.getContactDisplayInfo(findByAddress, number)
-        .then(function onData(data) {
-          data.source = 'contacts';
-          return data;
-        });
+      findByAddress(phoneNumber).then((contacts) => {
+        return Object.assign(
+          Utils.basicContact(phoneNumber, contacts),
+          { 
+            source: 'contacts' 
+          }
```

Comment:
nit: you can write on one line, I think it's more readable in this case: `{ source: 'contacts' }`